src/main/java/com/avispl/symphony/dal/avdevices/encoderdecoder/haivision/x4decoder/HaivisionX4DecoderCommunicator.java

```diff
@@ -69,13 +86,23 @@
  * @since 1.0.0
  */
 public class HaivisionX4DecoderCommunicator extends RestCommunicator implements Monitorable, Controller {
-
 	private AuthenticationCookie authenticationCookie;
 	private AuthenticationInfo authenticationInfo;
 	private HashMap<String, String> failedMonitor;
 	private Set<String> streamNameSet;
 	private Set<String> streamStatusSet;
 	private Set<String> portNumberSet;
+	private boolean isUpdateLocalDecoderControl = false;
+	private boolean isUpdateLocalStreamControl = false;
+
+	private boolean isEmergencyDelivery = false;
+	private ExtendedStatistics localExtendedStatistics;
+
+	// Decoder and stream DTO
+	private ArrayList<DecoderInfo> decoderInfoDTOList;
```

Comment:
It should be Upcasting as well
		private List<DecoderInfo> decoderInfoDTOList;

Reply:
I fixed it and the similar cases